applications/FluidDynamicsApplication/custom_processes/compute_one_side_nodal_pressure_gradient_process.cpp

```diff
@@ -0,0 +1,148 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Mohammad R. Hashemi
+//
+
+// Project includes
+#include "includes/define.h"
+#include "utilities/parallel_utilities.h"
+
+// Application includes
+#include "fluid_dynamics_application_variables.h"
+#include "compute_one_side_nodal_pressure_gradient_process.h"
+
+
+namespace Kratos
+{
+
+/* Public functions *******************************************************/
+
+/// constructor
+ComputeOneSideNodalPressureGradientProcess::ComputeOneSideNodalPressureGradientProcess(
+        ModelPart& rModelPart)
+    : Process(), mrModelPart(rModelPart) {}
+
+/// Constructor with Kratos parameters.
+ComputeOneSideNodalPressureGradientProcess::ComputeOneSideNodalPressureGradientProcess(
+    ModelPart& rModelPart,
+    Parameters Parameters)
+    : ComputeOneSideNodalPressureGradientProcess(
+        rModelPart
+    ){}
+
+/// Constructor with Kratos model
+ComputeOneSideNodalPressureGradientProcess::ComputeOneSideNodalPressureGradientProcess(
+    Model& rModel,
+    Parameters Parameters)
+    : ComputeOneSideNodalPressureGradientProcess(
+        rModel.GetModelPart(Parameters["model_part_name"].GetString())
+    ){}
+
+void ComputeOneSideNodalPressureGradientProcess::Execute(){
+
+    KRATOS_TRY;
+
+    // Set to zero
+    block_for_each(mrModelPart.Nodes(), [&](Node<3>& rNode){
+        rNode.SetValue(NODAL_AREA, 0.0);
+        rNode.FastGetSolutionStepValue(PRESSURE_GRADIENT).clear();
+    });
+
+    // Current domain size
+    const unsigned int num_dim = mrModelPart.GetProcessInfo()[DOMAIN_SIZE];
+    const unsigned int num_nodes = num_dim + 1; //For tetrahedra and triangles
+
+    // Auxiliar containers
+    double detJ0 = 0.0;
+    Matrix DN_DX, J0, InvJ0;
+    Vector N(num_nodes), pressures(num_nodes), distances(num_nodes), grad(num_dim);
+
+    // First element iterator
+    const auto it_element_begin = mrModelPart.ElementsBegin();
+
+    // Iterate over the elements
+    #pragma omp parallel for firstprivate(DN_DX, N, J0, InvJ0, detJ0, pressures, distances, grad)
+    for(int i_elem=0; i_elem<static_cast<int>(mrModelPart.Elements().size()); ++i_elem) {
```

Comment:
I'd create an auxiliary var to avoid doing the cast each time the exit condition of the loop is checked.